nodejs/dist/uws.js

```diff
@@ -209,9 +210,16 @@ class Server extends EventEmitter {
                 options.path = '/' + options.path;
             }
 
-            this.httpServer.on('upgrade', (request, socket, head) => {
-                if (!options.path || options.path == request.url.split('?')[0].split('#')[0]) {
-                    if (options.verifyClient) {
+            this._onServerUpgrade = (request, socket, head) => {
+
+                // check for wrong path
+                if (options.path) {
+                  var u = url.parse(request.url);
+                  if (u && u.pathname !== options.path)
+                    return;
```

Comment:
Francois, I agree with you. My quote was meant as sarcasm. I will most probably go with your design, but today I'm juat taking it slow.